### PR TITLE
1:1 screenshare tests

### PR DESCRIPTION
**Pull request by dbkr**
Fixes https://github.com/vector-im/element-call/issues/548

## Checklist

* [ ] Tests written for new code (and old code if feasible)
* [ ] Linter and other CI checks pass
* [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-js-sdk/blob/develop/CONTRIBUTING.md))

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: spec/unit/webrtc/call.spec.ts [hunk 1]
```diff
@@ -734,18 +733,16 @@ describe('Call', function() {
 
     describe("ignoring streams with ids for which we already have a feed", () => {
         const STREAM_ID = "stream_id";
-        const FEEDS_CHANGED_CALLBACK = jest.fn();
+        let FEEDS_CHANGED_CALLBACK;
```
Comment (robintown) [inline spec/unit/webrtc/call.spec.ts:736]: Type?

File: spec/unit/webrtc/call.spec.ts [hunk 2]
```diff
@@ -734,18 +733,16 @@ describe('Call', function() {
...
 
         beforeEach(async () => {
+            FEEDS_CHANGED_CALLBACK = jest.fn();
+
             await startVoiceCall(client, call);
             call.on(CallEvent.FeedsChanged, FEEDS_CHANGED_CALLBACK);
```
Comment (robintown) [inline spec/unit/webrtc/call.spec.ts:742]: This should have an accompanying `call.off`

File: spec/unit/webrtc/call.spec.ts [hunk 3]
```diff
@@ -941,4 +938,84 @@ describe('Call', function() {
...
+        });
+
+        it("enables screensharing", async () => {
+            await call.setScreensharingEnabled(true);
+
+            expect(call.feeds.filter(f => f.purpose == SDPStreamMetadataPurpose.Screenshare).length).toEqual(1);
+
+            client.client.sendEvent.mockReset();
+            const sendNegotiatePromise = new Promise<void>(resolve => {
+                client.client.sendEvent.mockImplementation(() => {
```
Comment (robintown) [inline spec/unit/webrtc/call.spec.ts:980]:
```suggestion
                client.client.sendEvent.mockImplementationOnce(() => {
```

File: spec/unit/webrtc/call.spec.ts [hunk 4]
```diff
@@ -941,4 +938,84 @@ describe('Call', function() {
...
+                }),
+            );
+        });
+
+        it("disables screensharing", async () => {
+            await call.setScreensharingEnabled(true);
+
+            client.client.sendEvent.mockReset();
+            const sendNegotiatePromise = new Promise<void>(resolve => {
+                client.client.sendEvent.mockImplementation(() => {
```
Comment (robintown) [inline spec/unit/webrtc/call.spec.ts:1008]:
```suggestion
                client.client.sendEvent.mockImplementationOnce(() => {
```

File: spec/unit/webrtc/call.spec.ts [hunk 5]
```diff
@@ -941,4 +938,84 @@ describe('Call', function() {
...
+                    resolve();
+                });
+            });
+
+            MockRTCPeerConnection.triggerAllNegotiations();
+            await sendNegotiatePromise;
+
+            await call.setScreensharingEnabled(false);
+
+            expect(call.feeds.filter(f => f.purpose == SDPStreamMetadataPurpose.Screenshare).length).toEqual(0);
```
Comment (robintown) [inline spec/unit/webrtc/call.spec.ts:1018]: I was going to suggest that this test assert that there was a screensharing feed to begin with, but then it probably makes more sense to just roll this and the previous test into one

Comment (dbkr) [reply spec/unit/webrtc/call.spec.ts:1008]: Yeah, this is probably a situation where this is clearer & easier & sticking too rigidly to the 'test once thing per test' mantra isn't the right thing to do.

**Review by robintown — APPROVED** [review]

**Pull request merged**